builder/lxc/step_lxc_create.go

```diff
@@ -17,51 +19,83 @@ type stepLxcCreate struct{}
 func (s *stepLxcCreate) Run(state multistep.StateBag) multistep.StepAction {
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
-
-	name := config.ContainerName
+	errorHandler := func(err error) {
+		state.Put("error", err)
+		ui.Error(err.Error())
+	}
 
 	// TODO: read from env
 	lxc_dir := "/var/lib/lxc"
+	name := config.ContainerName
 	rootfs := filepath.Join(lxc_dir, name, "rootfs")
 
-	if config.PackerForce || config.CleanupFirst {
-		s.destroy(config.ContainerName, ui)
+	if config.PackerForce {
+		s.destroy(name, ui)
 	}
 
-	commands := make([][]string, 3)
-	if config.CloneSource == "" {
-		commands[0] = append(config.EnvVars, []string{"lxc-create", "-n", name, "-t", config.Name, "--"}...)
-		commands[0] = append(commands[0], config.Parameters...)
+	if config.LxcTemplate.Name != "" {
+		commands := make([][]string, 2)
+		commands[0] = append(config.LxcTemplate.EnvVars, []string{"lxc-create", "-n", name, "-t", config.LxcTemplate.Name, "--"}...)
+		commands[0] = append(commands[0], config.LxcTemplate.Parameters...)
+		// prevent tmp from being cleaned on boot, we put provisioning scripts there
+		// TODO: wait for init to finish before moving on to provisioning instead of this
+		commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
+
+		ui.Say("Creating container from template...")
+		for _, command := range commands {
+			log.Printf("Executing sudo command: %#v", command)
+			if err := s.SudoCommand(command...); err != nil {
+				errorHandler(fmt.Errorf("Command (%s) failed with error: %s", strings.Join(command, " "), err))
```

Comment:
Shouldn't we be passing back the inciting error with just `errorHandler(err)`? We'll have a stack trace of where it's coming from so I don't know what a personalized message helps with and it might obscure important data from the error passed down.